src/common/utils.js

```diff
@@ -205,6 +198,14 @@ export const listenEvents = async (api, section, method, timeout = undefined) =>
     };
 
     listenSystemEvents().catch(console.log);
+
+    // Set timeout to stop event listening.
+    if (timeout) {
+        timeoutId = setTimeout(() => {
+            unsub();
+            resolve(false);
+        }, timeout);
+    }
```

Comment:
1. `timeout` should be required, instead of `timeout = undefined` for this task. I don’t see a reason why the caller doesn’t set timeout.
2. What is the reason of moving this block from top to bottom? What problem does the code chagne solve?